test/common/Promise.spec.ts

```diff
@@ -126,11 +136,14 @@ describe(
           expect(queue.length).toEqual(1);
           expect(log).toEqual([]);
           flush();
-          expect(log).toEqual(['RValue', 'second value']);
+          setTimeout(() => {
```

Comment:
yes, just like the above one, I am just began to debug those tests, I will finish modifying those tests and let you review.